src/constants.ts

```diff
@@ -1,62 +1,56 @@
-export const sc_type_edge_perm = 0x800;
-
-// struct node types
-export const sc_type_node_tuple = 0x80;
-export const sc_type_node_struct = 0x100;
-export const sc_type_node_role = 0x200;
-export const sc_type_node_norole = 0x400;
-export const sc_type_node_class = 0x800;
-export const sc_type_node_abstract = 0x1000;
-export const sc_type_node_material = 0x2000;
-
-export const sc_type_arc_pos_const_perm =
-  sc_type_edge_access | sc_type_const | sc_type_edge_pos | sc_type_edge_perm;
-export const sc_type_arc_pos_var_perm =
-  sc_type_edge_access | sc_type_var | sc_type_edge_pos | sc_type_edge_perm;
-
-// masks
-export const sc_type_element_mask =
-  sc_type_node |
-  sc_type_link |
-  sc_type_uedge_common |
-  sc_type_dedge_common |
-  sc_type_edge_access;
-
-export const sc_type_constancy_mask = sc_type_const | sc_type_var;
-
-export const sc_type_positivity_mask =
-  sc_type_edge_pos | sc_type_edge_neg | sc_type_edge_fuz;
-
-export const sc_type_permanency_mask = sc_type_edge_perm | sc_type_edge_temp;
-
-export const sc_type_node_struct_mask =
-  sc_type_node_tuple |
-  sc_type_node_struct |
-  sc_type_node_role |
-  sc_type_node_norole |
-  sc_type_node_class |
-  sc_type_node_abstract |
-  sc_type_node_material;
-
-export const sc_type_edge_mask =
-  sc_type_edge_access | sc_type_dedge_common | sc_type_uedge_common;
+export const sc_type_const = 0x20
+export const sc_type_var = 0x40
+
+// sc-arc actuality
+export const sc_type_actual_arc = (sc_type_membership_arc | 0x1000)
+export const sc_type_inactual_arc = (sc_type_membership_arc | 0x2000)
+
+// sc-arc permanence
+export const sc_type_temp_arc = (sc_type_membership_arc | 0x400)
+export const sc_type_perm_arc = (sc_type_membership_arc | sc_type_actual_arc | 0x800)
+
+// sc-arc positivity
+export const sc_type_pos_arc = (sc_type_membership_arc | 0x80)
+export const sc_type_neg_arc = (sc_type_membership_arc | 0x100)
+export const sc_type_fuz_arc = (sc_type_membership_arc | 0x200)
+
+// semantic sc-node types
+export const sc_type_node_link = (sc_type_node | 0x2)
+export const sc_type_node_tuple = (sc_type_node | 0x80)
+export const sc_type_node_structure = (sc_type_node | 0x100)
+export const sc_type_node_role = (sc_type_node | 0x200)
+export const sc_type_node_norole = (sc_type_node | 0x400)
+export const sc_type_node_class = (sc_type_node | 0x800)
+export const sc_type_node_superclass = (sc_type_node | 0x1000)
+export const sc_type_node_material = (sc_type_node | 0x2000)
+
+// type mask
+export const sc_type_element_mask = (sc_type_node | sc_type_connector)
+export const sc_type_connector_mask = (sc_type_common_edge | sc_type_common_arc | sc_type_membership_arc)
+export const sc_type_arc_mask = (sc_type_common_arc | sc_type_membership_arc)
+
+export const sc_type_constancy_mask = (sc_type_const | sc_type_var)
+export const sc_type_actuality_mask = (sc_type_actual_arc | sc_type_inactual_arc)
+export const sc_type_permanency_mask = (sc_type_perm_arc | sc_type_temp_arc)
```

Comment:
this one makes actuality a part of permanency, but inactuality is not part of permanency
just warning about inconsistency, probably sc-machine has same issue

Reply:
all good here

Reply:
in checking type extendability for actualArc.isExtendableTo(temporalArc):
1. first arc is actual, second arc has no actuality, false isn't returned
2. first arc has no permanency(but comparing with this mask says that it has actual permanency), second arc has temporal permanency, false **is**(!) returned
but it should be allowed

Reply:
Yes, there is a problem, but not with what you said. If the first arc is temporal without specified actuality and the second arc is actual, true isn't returned. I fixed this in sc-machine commit https://github.com/ostis-ai/sc-machine/pull/426/commits/059f5779ca972effbe10c406961271b2d38f2d77